nepalingo-web/src/components/StreakContext.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  ReactNode,
+} from "react";
+import supabase from "./userAuth/supabaseClient";
+import { useAuth } from "./userAuth/AuthContext";
+
+interface StreakContextProps {
+  currentStreak: number;
+  longestStreak: number;
+  updateStreak: () => void;
+}
+
+export const StreakContext = createContext<StreakContextProps>({
+  currentStreak: 0,
+  longestStreak: 0,
+  updateStreak: () => {},
+});
+
+export const StreakProvider = ({ children }: { children: ReactNode }) => {
+  const { user } = useAuth();
+  const [currentStreak, setCurrentStreak] = useState(1);
+  const [longestStreak, setLongestStreak] = useState(1);
+
+  useEffect(() => {
+    const fetchStreakData = async () => {
+      if (user) {
+        const { data } = await supabase
+          .from("user_daily_streaks")
+          .select("*")
+          .eq("user_id", user.id)
+          .single();
+
+        if (data) {
+          setCurrentStreak(data.current_streak);
+          setLongestStreak(data.longest_streak);
+        }
+      }
+    };
+
+    fetchStreakData();
```

Comment:
I think you can get rid of the fetchStreakData function here and just have your code directly in the useEffect

Reply:
can I ask about the reason? I mean....having the fetch function is easier to read and test and it can be reused too. ( If we ever need to fetch the streak data from another part of our application, we can reuse this function.) That's what I thought...
are there any better cons of using it directly in useEffect?

Reply:
We talked about this and you can't do that actually, my bad.